File: Java/algoritmo007/Main.java
package algoritmo007;

import java.util.Scanner;

public class Main {
    public static void executar(){
        Scanner scanner = new Scanner(System.in);

        System.out.println("Ler a prova.");
        System.out.print("Quantas questões tem na prova? ");

        int numeroDeQuestoes = Integer.parseInt(scanner.nextLine());
        int contadorDeQuestoes = 1;
        int tempoParaProva = 60;

        while (numeroDeQuestoes > 0 && tempoParaProva > 0){
            System.out.print("Você sabe responder a %dª questão? (s/n)".formatted(contadorDeQuestoes));
            String resposta = scanner.nextLine();

            if(resposta.equalsIgnoreCase("s")){
                System.out.println("Questão: %d respondida.".formatted(contadorDeQuestoes));
                tempoParaProva -= 3;
            } else {
                System.out.println("Você pulou a questão: %d.".formatted(contadorDeQuestoes));
                tempoParaProva -= 1;
            }

            contadorDeQuestoes++;
            numeroDeQuestoes--;
        }

        if (tempoParaProva <= 0) {

            System.out.println("O tempo acabou.");
            System.out.println("Entregar a prova.");

        } else if (numeroDeQuestoes <= 0){

            System.out.println("Você terminou a prova.");
            System.out.println("Entregar a prova.");
            
        }

    }

    public static void main(String[] args){
        executar();
    }
}

File: Python/algoritmo007/main.py
def executar():
    print("Ler a prova.")

    numero_de_questoes = int(
        input("Quantas questões tem na prova? ")
    )

    contador_de_questoes = 1
    tempo_para_prova = 60

    while numero_de_questoes > 0 and tempo_para_prova > 0:

        resposta = input(
            f"Você sabe responder a {contador_de_questoes}ª questão? (s/n): "
        )

        if resposta.lower() == "s":
            print(
                f"Questão: {contador_de_questoes} respondida"
            )

            tempo_para_prova -= 3

        else:
            print(
                f"Você pulou a questão: {contador_de_questoes}"
            )

            tempo_para_prova -= 1

        contador_de_questoes += 1
        numero_de_questoes -= 1

    if tempo_para_prova <= 0:
        print("O tempo acabou.")
        print("Entregar a prova.")

    elif numero_de_questoes <= 0:
        print("Você terminou a prova.")
        print("Entregar a prova.")


executar()